lib/ansible/module_utils/redfish_utils.py

```diff
@@ -322,42 +322,52 @@ def get_disk_inventory(self):
         controller_list = []
         disk_results = []
         # Get these entries, but does not fail if not found
-        properties = ['Name', 'Manufacturer', 'Model', 'Status', 'CapacityBytes']
+        properties = ['BlockSizeBytes', 'CapableSpeedGbs', 'CapacityBytes',
+                      'EncryptionAbility', 'EncryptionStatus',
+                      'FailurePredicted', 'HotspareType', 'Id', 'Identifiers',
+                      'Manufacturer', 'MediaType', 'Model', 'Name',
+                      'PartNumber', 'PhysicalLocation', 'Protocol', 'Revision',
+                      'RotationSpeedRPM', 'SerialNumber', 'Status']
 
         # Find Storage service
         response = self.get_request(self.root_uri + self.systems_uri)
         if response['ret'] is False:
             return response
         data = response['data']
 
-        if 'SimpleStorage' not in data:
```

Comment:
It is valid for a ComputerSystem to have a `Storage` resource or a `SimpleStorage` resource or both. And each of these can describe Drives in the system. So we should keep the original `SimpleStorage` logic along with your newly added `Storage` logic. If there are both `SimpleStorage` and `Stroage` resoures present, the returned `result` should include the drive info from both.

Something like this:

```
    if 'SimpleStorage' not in data and 'Storage' not in data:
        return {'ret': False, 'msg': "SimpleStorage or Storage resource not found"}
    if 'SimpleStorage' in data:
        # do the original SimpleStorage logic
    if 'Storage' in data:
        # do the new Storage logic
    ```

Reply:
@billdodd, should this approach be applied to https://github.com/ansible/ansible/pull/52928 as well?

Reply:
@jose-delarosa - No, I asked Mike R. about this and he said that for Storage Controllers it should just look at `Storage`.

Reply:
Gotcha, I can tackle this change.

Reply:
This should be resolved now.